### ISSUE TITLE
OpenAPI 3 spec format incorrect

### ISSUE THREAD

**Issue by WoH** [lukeautry/tsoa#404]
Tl:Dr: The OpenAPI 3 spec generated contains errors. For better reproducibility, I used the spec generated from the tests to illustrate some of the issues:

```
    "/GetTest/{numberPathParam}/{booleanPathParam}/{stringPathParam}": {
      "get": {
        "operationId": "GetModelByParams",
        "produces": [
          "application/json"
        ],
        "responses": {
          "200": {
            "description": "Ok",
            "schema": {
              "$ref": "#/components/schemas/TestModel"
            }
          }
        },
        "security": [],
        "parameters": [
          {
            "description": "This is a description for numberPathParam",
            "in": "path",
            "name": "numberPathParam",
            "required": true,
            "format": "double",
            "type": "number",
            "minimum": 1,
            "maximum": 10
          },
-----------------SNIP------------------
        ]
      }
    },
```

should look like this, according to the OAPI3 specification:

```
    "/GetTest/{numberPathParam}/{booleanPathParam}/{stringPathParam}": {
      "get": {
        "operationId": "GetModelByParams",
        "responses": {
          "200": {
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/TestModel"
                }
              }
            },
            "description": "Ok"
          }
        },
        "security": [],
        "parameters": [
          {
            "description": "This is a description for numberPathParam",
            "in": "path",
            "name": "numberPathParam",
            "required": true,
            "schema": {
              "format": "double",
              "type": "number",
              "minimum": 1,
              "maximum": 10
            }
          },
-----------------SNIP------------------

        ]
      }
    },
```

Notice how `produces` is no longer valid, but the value moved inside the response as `content`,
the parameter schema is now its own thing inside a parameter object, and, not shown in this sample, the examples have to be moved as well.

I am sure there are some more things I have not yet discovered, but I opened  #401 that produces a valid OpenAPI3 document for the entire GetController.

I would be glad if @endor could take a look?
